### ISSUE TITLE
Tensor Padding Element

### ISSUE THREAD

**Issue by myungjoo** [nnstreamer/nnstreamer#4106]
Related w/ #4105

This may be implemented in ```tensor-transform``` or created as an independent element, ```tensor-padding```.

Example usage: (the property syntax doen't matter. the implementer should design it, too)
``` ... ! ( 5:3:1:1 tensor ) ! tensor-padding value=0 left=3,2 right=2,2 dim=0,1 ! (10:7:1:1 tensor ) ! ...```
The additional entires are filled by 0.

**Comment by niley7464** [comment]
I have a question about supported dimension in pad element and how to implement it.

tensorflow : while loop with vector
```cpp
 static bool CollapseAdjacentNonPaddedDimensions(
...
    while (i < paddings.dimension(0)) {
      if (paddings(i, 0) != 0 || paddings(i, 1) != 0) {
        // If padded, copy the original dimension over.
        collapsed_input_shape->InsertDim(collapsed_input_shape->dims(),
                                         input_shape.dim_size(i));
        collapsed_output_shape->InsertDim(collapsed_output_shape->dims(),
                                          output_shape.dim_size(i));
        collapsed_paddings.push_back({paddings(i, 0), paddings(i, 1)});
        ++i;
```
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/kernels/pad_op.cc#L157

numpy : for loop with python zip function
```py
def _pad_simple(array, pad_width, fill_value=None):
...
    original_area_slice = tuple(
        slice(left, left + size)
        for size, (left, right) in zip(array.shape, pad_width)
    )
```
https://github.com/numpy/numpy/blob/v1.26.0/numpy/lib/arraypad.py#L86

pytorch and nntrainer : supports only some ranks
```cpp
void cpu_padding((
...
  if (ndim == 1) {
...
    });
  } else if (ndim == 2) {
...
    });
  } else if (ndim == 3) {
...
    });
  } else {
    TORCH_INTERNAL_ASSERT(false, "expect input dim to be 1d, 2d or 3d.");
  }
```

```c
class Padding2D final : public nntrainer::Property<std::string> {
public:
...
```
https://github.com/pytorch/pytorch/blob/main/aten/src/ATen/native/cpu/PaddingKernel.cpp#L180
https://github.com/nnstreamer/nntrainer/blob/main/nntrainer/layers/common_properties.h#L422
(In pytorch documentation, it says `Constant padding is implemented for arbitrary dimensions.`. But I think rank higher than 3 is implemented differently.)

If tensor padding element supports N dimension (N<=max rank of tensor), I think loop macro or recursive function are needed.
So how about just supports only 2d, 3d dimension first like tensor transpose?
tensor transpose in tensor transfrom is now fixed to 4

**Comment by myungjoo** [comment]
Try to use ORC, refer to https://github.com/nnstreamer/nnstreamer/blob/main/gst/nnstreamer/elements/gsttensor_transform.c and https://github.com/nnstreamer/nnstreamer/blob/main/gst/nnstreamer/elements/nnstreamer-orc.orc

**Comment by myungjoo** [comment]
Closing: implemented. tensor_transform now has a padding mode (commit 0985d731, Jan 2024) supporting left/right/top/bottom/front/back padding with NCHW/NHWC layout options.
